cluster-autoscaler/provisioningrequest/orchestrator/orchestrator.go

```diff
@@ -0,0 +1,149 @@
+	"k8s.io/autoscaler/cluster-autoscaler/clusterstate"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+	"k8s.io/autoscaler/cluster-autoscaler/estimator"
+	"k8s.io/autoscaler/cluster-autoscaler/processors/status"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/apis/autoscaling.x-k8s.io/v1beta1"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/checkcapacity"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/conditions"
+	provreq_pods "k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/pods"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqclient"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/scheduling"
+	ca_errors "k8s.io/autoscaler/cluster-autoscaler/utils/errors"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/taints"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	ca_processors "k8s.io/autoscaler/cluster-autoscaler/processors"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+type scaleUpMode interface {
+	ScaleUp([]*apiv1.Pod, []*apiv1.Node, []*appsv1.DaemonSet,
+		map[string]*schedulerframework.NodeInfo) (*status.ScaleUpStatus, ca_errors.AutoscalerError)
+	Initialize(*context.AutoscalingContext, *ca_processors.AutoscalingProcessors, *clusterstate.ClusterStateRegistry,
+		estimator.EstimatorBuilder, taints.TaintConfig, *scheduling.HintingSimulator)
+}
+
+// provReqOrchestrator is an orchestrator that contains orchestrators for all supported Provisioning Classes.
+type provReqOrchestrator struct {
+	initialized  bool
+	context      *context.AutoscalingContext
+	client       provreqclient.ProvisioningRequestClient
+	injector     *scheduling.HintingSimulator
+	scaleUpModes []scaleUpMode
+}
+
+// New return new orchestrator.
+func New(kubeConfig *rest.Config) (*provReqOrchestrator, error) {
+	client, err := provreqclient.NewProvisioningRequestClient(kubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &provReqOrchestrator{client: client, scaleUpModes: []scaleUpMode{checkcapacity.New(client)}}, nil
+}
+
+// Initialize initialize orchestrator.
+func (o *provReqOrchestrator) Initialize(
+	autoscalingContext *context.AutoscalingContext,
+	processors *ca_processors.AutoscalingProcessors,
+	clusterStateRegistry *clusterstate.ClusterStateRegistry,
+	estimatorBuilder estimator.EstimatorBuilder,
+	taintConfig taints.TaintConfig,
+) {
+	o.initialized = true
+	o.context = autoscalingContext
+	o.injector = scheduling.NewHintingSimulator(autoscalingContext.PredicateChecker)
+	for _, mode := range o.scaleUpModes {
+		mode.Initialize(autoscalingContext, processors, clusterStateRegistry, estimatorBuilder, taintConfig, o.injector)
+	}
+}
+
+// ScaleUp run ScaleUp for each Provisionining Class. As of now, CA pick one ProvisioningRequest,
+// so only one ProvisioningClass return non empty scaleUp result.
+// In case we implement multiple ProvisioningRequest ScaleUp, the function should return combined status
+func (o *provReqOrchestrator) ScaleUp(
```

Comment:
Please make this (and other similar places) one line: https://google.github.io/styleguide/go/guide.html#line-length.

Reply:
The line will be too long. The style guide suggest "If a line feels too long, prefer refactoring instead of splitting it.", but refactoring would complicate this PR.
If we aim readability first, then the current format is more readable than the long line.